synapse/rest/client/v2_alpha/report_event.py

```diff
@@ -42,15 +38,14 @@ async def on_POST(self, request, room_id, event_id):
         user_id = requester.user.to_string()
 
         body = parse_json_object_from_request(request)
-        assert_params_in_dict(body, ("reason", "score"))
 
-        if not isinstance(body["reason"], str):
+        if not isinstance(body.get("reason", ""), str):
```

Comment:
Note that this will fail if it's `None` (if you want to support it you probably want to do `body.get("reason") or ""` instead). Same comment for `score` a few lines down.

Reply:
Ah ok. So this is an acceptable request?
```
{
    "reason": null,
    "score": -100
}
```

Reply:
Sorry, I got things mixed up. Your solution is correct.